ONSdigital/ras-frontstage
# Move seft upload

# What and why?

Previously we would get the file from the user and make a request to collection-instrument to actually act on it.  Now we'll just act on it ourselves by putting the file in the bucket and the message into pubsub.

The code around this was.... not great, so as well as copying the code across from collection-instrument, it's been refactored so it should be more efficient and easier to maintain.

As well as this, I've moved a bunch of tests that don't rely on any external sources from integration to unit.  The theory was that integration tests (that need and/or test real backing services) would live in their own section, but there are a lot of tests in the integration folder that don't rely on anything external (because of mocking, etc)

# How to test?

- Deploy this to your namespace
- Run SEFT acceptance tests (easily done through spinnaker)
- Log in as user and try and upload a SEFT response.
- Check in the bucket in your namespace that it's appeared as expected and that a pubsub message was put onto the queue.

You'll notice there is a lot of spammy logging, it's because we run the app at DEBUG level in development, but we run it at INFO level further downstream in preprod and prod so that'll go away.  You can verify this yourself by setting the LOGGING_LEVEL in your environment.

# Trello
